notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java

```diff
@@ -1466,22 +1469,46 @@ public DraftPage getLatestDraftPageByUserAndTargetPageAndLang(Long targetPageId,
    * {@inheritDoc}
    */
   @Override
-  public void deleteVersionsByNoteIdAndLang(Long noteId, String lang) throws WikiException {
+  public void deleteVersionsByNoteIdAndLang(Long noteId, String lang) throws Exception {
+    Page note = getNoteById(String.valueOf(noteId));
+    if (note != null) {
+      deleteNoteMetadataProperties(note, lang, NOTE_METADATA_PAGE_OBJECT_TYPE);
```

Comment:
I prefer that the properties deletion come after the version deletion itself, so when there is a problem with the version deletion the properties will not be deleted

Reply:
https://github.com/Meeds-io/notes/pull/1094#discussion_r1753509503